Sources/StreamChatUI/CommonViews/MessageInputView/ChatMessageInputView.swift

```diff
@@ -24,44 +45,69 @@ open class _ChatMessageInputView<ExtraData: ExtraDataTypes>: _View, ComponentsPr
         .commandLabel.init()
         .withoutAutoresizingMaskConstraints
 
-    /// A button to clean the current typing information
-    public private(set) lazy var cleanButton: UIButton = UIButton()
+    /// A button to clear the current typing information.
+    public private(set) lazy var clearButton: UIButton = components
+        .closeButton.init()
         .withoutAutoresizingMaskConstraints
 
     override open func setUpAppearance() {
         super.setUpAppearance()
 
-        let cleanButtonImage = appearance.images.closeCircleTransparent.tinted(with: appearance.colorPalette.inactiveTint)
-        cleanButton.setImage(cleanButtonImage, for: .normal)
+        let closeTransparentImage = appearance.images.closeCircleTransparent
+            .tinted(with: appearance.colorPalette.inactiveTint)
+        clearButton.setImage(closeTransparentImage, for: .normal)
+
+        container.clipsToBounds = true
+        container.layer.cornerRadius = 18
+        container.layer.borderWidth = 1
+        container.layer.borderColor = appearance.colorPalette.border.cgColor
     }
     
     override open func setUpLayout() {
         addSubview(container)
         container.pin(to: layoutMarginsGuide)
-        directionalLayoutMargins.leading = 8
-        directionalLayoutMargins.top = 1
-        directionalLayoutMargins.trailing = 8
-        directionalLayoutMargins.bottom = 1
+        directionalLayoutMargins.leading = 0
+        directionalLayoutMargins.top = 0
+        directionalLayoutMargins.trailing = 0
+        directionalLayoutMargins.bottom = 0
 
-        container.preservesSuperviewLayoutMargins = false
-        container.alignment = .center
-        container.spacing = 4
+        container.isLayoutMarginsRelativeArrangement = true
+        container.directionalLayoutMargins.leading = 4
+        container.directionalLayoutMargins.top = 0
+        container.directionalLayoutMargins.trailing = 4
+        container.directionalLayoutMargins.bottom = 0
```

Comment:
```swift
container.directionalLayoutMargins = .init(top: 0, leading: 4, bottom: 0, trailing: 4)
```
Maybe this style is a bit better? Especially when you're changing all values there.

Reply:
Done ✅